octobot_tentacles_manager/managers/tentacles_setup_manager.py

```diff
@@ -0,0 +1,139 @@
+#  Drakkar-Software OctoBot-Tentacles-Manager
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+from asyncio import gather
+from os.path import join, isfile, exists
+from shutil import rmtree
+
+import aiofiles
+
+from octobot_commons.logging.logging_util import get_logger
+from octobot_tentacles_manager.configuration.tentacles_setup_configuration import TentaclesSetupConfiguration
+from octobot_tentacles_manager.constants import USER_TENTACLE_SPECIFIC_CONFIG_PATH, PYTHON_INIT_FILE, \
+    TENTACLES_FOLDERS_ARCH, DEFAULT_TENTACLE_CONFIG, TENTACLES_PATH, USER_TENTACLE_CONFIG_PATH, \
+    TENTACLES_REQUIREMENTS_INSTALL_TEMP_DIR
+from octobot_tentacles_manager.util.file_util import find_or_create
+from octobot_tentacles_manager.util.tentacle_explorer import load_tentacle_with_metadata
+
+
+class TentaclesSetupManager:
+
+    TENTACLE_IMPORT_HEADER = """from octobot_tentacles_manager.api.inspector import check_tentacle_version
+from octobot_commons.logging.logging_util import get_logger
+"""
+
+    def __init__(self, tentacle_path):
+        self.tentacle_path = tentacle_path
+        self.default_tentacle_config = DEFAULT_TENTACLE_CONFIG
+
+    async def refresh_tentacles_config_file(self):
+        available_tentacle = await load_tentacle_with_metadata(self.tentacle_path)
+        tentacle_setup_config = TentaclesSetupConfiguration()
+        await tentacle_setup_config.read_config()
+        await tentacle_setup_config.fill_tentacle_config(available_tentacle, self.default_tentacle_config)
+        await tentacle_setup_config.save_config()
+
+    async def create_missing_tentacles_arch(self):
+        # tentacle user config folder
+        await find_or_create(USER_TENTACLE_SPECIFIC_CONFIG_PATH)
+        # tentacles folder
+        found_existing_installation = not await find_or_create(self.tentacle_path)
+        # tentacle mail python init file
+        await find_or_create(join(self.tentacle_path, PYTHON_INIT_FILE), False,
+                             self.get_module_init_file_content(TENTACLES_FOLDERS_ARCH.keys()))
+        # tentacle inner architecture
+        await self._rec_create_missing_files(self.tentacle_path, TENTACLES_FOLDERS_ARCH)
+        return found_existing_installation
+
+    @staticmethod
+    async def update_tentacle_type_init_file(tentacle, target_tentacle_path, remove_import=False):
```

Comment:
Why this is not made by a Tentacle specific class ?

Reply:
This function updates the python init file of the given tentacle type (Evaluator/TA, Trading/Mode, etc), not the init file from the tentacle itself. Therefore it edits a file from the current tentacles setup (common with other tentacles of the same type) and not a specific tentacle. To me it makes more sence in this class.